### PR TITLE
Enforce smaller block size before DRBD migration starts

### PR DESCRIPTION
**Pull request by TuxPowered42**
Problem on VM:
```
XFS (vda): xfs_log_force: error 5 returned.
```

Problem on HV:
```
bio too big device drbd0 (16 > 8)
```

### CODE REVIEW

File: igvm/hypervisor.py [hunk 1]
```diff
@@ -536,6 +536,7 @@ def migrate_vm(
                         'DRBD migration is supported only between hypervisors '
                         ' using LVM storage!'
                     )
+                vm.safe_blocksize()
```
Comment (juergenthomann) [inline igvm/hypervisor.py]: I think we should find a different name for it. It currently could look like a getter (get me the safe blocksize for this vm). We already have some more methods starting with set_, so maybe use set_safe_blocksize()

File: igvm/vm.py [hunk 2]
```diff
@@ -500,3 +500,15 @@ def unblock_autostart(self):
...
+    def safe_blocksize(self):
+        """ Reduce maximum number of KiB allowed for FS to request from
+            block layer
+
+        This is required for DRBD storage migrations. During disk migration
+        a DRBD will consist of devices on multiple hypervisors which might have
+        different size. It was observed that during high IO requests were
+        sometimes rejected by block layer causing filesystem to crash.
+        """
+        self.run('echo 64 > /sys/class/block/vda/queue/max_sectors_kb')
```
Comment (juergenthomann) [inline igvm/vm.py]: Do checked for the lowest number? Maybe 128 is low enough for us.

File: igvm/hypervisor.py (hunk 1 above)
Comment (TuxPowered42) [reply igvm/hypervisor.py]: Sure, I can fix that.

**Comment by TuxPowered42** [comment]
retest this please

**Comment by TuxPowered42** [comment]
retest this please

File: igvm/vm.py [hunk 3]
```diff
@@ -500,3 +500,16 @@ def unblock_autostart(self):
...
+        """ Reduce maximum number of KiB allowed for FS to request from
+            block layer
+
+        This is required for DRBD storage migrations. During disk migration
+        a DRBD will consist of devices on multiple hypervisors which might have
+        different maximum block size. It was observed that during high IO
+        requests were sometimes rejected by block layer causing filesystem to
+        crash.
+        """
+        self.run('echo 64 > /sys/class/block/vda/queue/max_sectors_kb')
```
Comment (Woellchen) [inline igvm/vm.py]: can we not query correct block size from the hvs taking part in the migration?

Comment (HerrSpace) [reply igvm/vm.py]: From all the information I have I can't really follow how you even conclusively found out it's related to max_sector_kb. Maybe the write just fails at the moment we switch to the drbd device?

Comment (juergenthomann) [reply igvm/vm.py]: It keeps failing, so it does not recover from an error until the server is killed.

File: igvm/hypervisor.py [hunk 4]
```diff
@@ -536,6 +536,23 @@ def migrate_vm(
...
+                        vm.hypervisor.get_volume_by_vm(vm).path()
+                    )
+                    dst_block_size = target_hypervisor.get_block_size(
+                        target_hypervisor.get_volume_by_vm(vm).path()
+                    )
+                    log.debug(
+                        'Block sizes: VM {}, Source HV {}, Destination HV {}'
+                        .format(vm_block_size, src_block_size, dst_block_size)
+                    )
+                    vm.set_block_size('vda', min(
```
Comment (Woellchen) [inline igvm/hypervisor.py]: awesome

**Pull request merged**